src/main/java/org/jenkinsci/plugins/ewm/steps/ExwsAllocateExecution.java

```diff
@@ -122,6 +142,28 @@ protected ExternalWorkspace run() throws Exception {
     }
 
     /**
+     * Gets the environment variables based on the current run.
+     *
+     * @return the {@link EnvVars} for the current run
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    private EnvVars getEnvVars() throws IOException, InterruptedException {
```

Comment:
Mostly what I did was to copy/paste from the Copy Artifact plugin. It had some additional comments in there. I quote:
> ```java
// Abstract#getEnvironment(TaskListener) put build parameters to
// environments, but Run#getEnvironment(TaskListener) doesn't.
// That means we can't retrieve build parameters from WorkflowRun
// as it is a subclass of Run, not of AbstractBuild.
// We need expand build parameters manually.
// See JENKINS-26694, JENKINS-30357 for details.
for(ParametersAction pa: build.getActions(ParametersAction.class)) {
    // We have to extract parameters manally as ParametersAction#buildEnvVars
    // (overrides EnvironmentContributingAction#buildEnvVars)
    // is applicable only for AbstractBuild.
    for(ParameterValue pv: pa.getParameters()) {
        pv.buildEnvironment(build, env);
    }
}
```

Anyway, if we migrate the *BuildSelector* to a new repo, the new signature should not have EnvVars as parameter, so this may not be needed anymore.